### DISCUSSION TITLE
Call async request several times with the same AsyncClient with asyncio.run() is causing RuntimeError: Event loop is closed

### DISCUSSION THREAD

**Discussion by PFaurel** [encode/httpx#2489] [Potential Issue] [answered]
When AsyncClient is used several times with an async function that's sending an http request, the first time is successful, all other attempts are failing with RuntimeError exception and "Event loop is closed" error message.

To reproduce:

import httpx

if __name__ == '__main__':

    client = httpx.AsyncClient()

    async def home():
        r = await client.get('https://www.example.com/')
        return r

    asyncio.run(home())
    asyncio.run(home())  # => RuntimeError: Event loop is closed
    asyncio.run(home())  # => RuntimeError: Event loop is closed
    asyncio.run(home())  # => RuntimeError: Event loop is closed

**Comment by lovelydinosaur** [comment, marked as answer]
So, yeah...

Don't try to share an `AsyncClient` across multiple event loops. That won't work, and I'd assume we probably *wouldn't be able* to make it work.

You either want...

```python
async def home():
    async with httpx.AsyncClient() as client:
        r = await client.get('https://www.example.com/')
    return r

# Not sure why you'd want to do this, but at least it's not broken now...
asyncio.run(home())
asyncio.run(home())
asyncio.run(home())
asyncio.run(home())
```

Or more sensibly...

```python
async def home(client):
    r = await client.get('https://www.example.com/')
    return r

async def main():
    # Structured correctly.
    # Single client shared across multiple requests.
    async with httpx.Client() as client:
        await home(client)
        await home(client)
        await home(client)
        await home(client)

asyncio.run(main())
```

**Reply by lovelydinosaur** [reply]
Punted this to a discussion.

If someone thinks we should either have some different behaviour or different documentation here, then let's discuss that, and then consider opening an issue.
